2_process/src/do_lake_toha_tasks.R

```diff
@@ -53,13 +51,37 @@ create_lake_tasks <- function(task_df_fn, log_folder){
     } 
   )
   
-  # ---- combine into a task plan ---- #
+  ##-- Create the task plan --##
   
-  gif_task_plan <- scipiper::create_task_plan(
+  task_plan <- create_task_plan(
     task_names=tasks$site_id,
     task_steps=list(split_morphometry,
-      calculate_model_toha),
+                    calculate_model_toha),
     add_complete=FALSE,
-    final_steps=c(toha_step_name),
-    ind_dir='2_process/log')
+    final_steps=c(toha_step_name))
+  
+  ##-- Create the task remakefile --##
+  
+  task_makefile <- sprintf('2_%s_lake_tasks.yml', model_prefix)
+  create_task_makefile( 
+    task_plan=task_plan,
+    makefile=task_makefile,
+    include='remake.yml',
+    sources=c(...),
+    packages=c("purrr", "dplyr", "mda.lakes", "feather", "rLakeAnalyzer"),
+    final_targets = final_target,
+    as_promises = TRUE,
+    tickquote_combinee_objects = TRUE
+  )
+  
+  ##-- Build the tasks --##
+  
+  loop_tasks(task_plan = task_plan,
+             task_makefile = task_makefile,
+             num_tries = 1, 
+             n_cores = n_cores)
+  
+  # Now return the file name of the final target
+  return(remake::fetch(sprintf("%s_promise", basename(final_target)), remake_file=task_makefile))
```

Comment:
see comment below about fetching the promise vs returning the `final_target` filepath.

Reply:
and comment below about not returning anything when a recipe target is a file